catalogs/catalog-hive/src/main/java/com/datastrato/gravitino/catalog/hive/HiveCatalogOperations.java

```diff
@@ -135,6 +158,91 @@ public void initialize(Map<String, String> conf) throws RuntimeException {
     // and gravitinoConfig will be passed to Hive config, and gravitinoConfig has higher priority
     mergeConfig.forEach(hadoopConf::set);
     hiveConf = new HiveConf(hadoopConf, HiveCatalogOperations.class);
+    UserGroupInformation.setConfiguration(hadoopConf);
+
+    if (UserGroupInformation.AuthenticationMethod.KERBEROS
+        == SecurityUtil.getAuthenticationMethod(hadoopConf)) {
+      try {
+        File keyTabFile = new File(String.format("/tmp/gravitino-%s-keytab", entity.id()));
+        keyTabFile.deleteOnExit();
+        if (keyTabFile.exists() && !keyTabFile.delete()) {
+          LOG.warn("Fail to delete key tab file {}", keyTabFile.getAbsolutePath());
+        }
+
+        String keyTabUri =
+            (String)
+                catalogPropertiesMetadata.getOrDefault(conf, HiveCatalogPropertiesMeta.KET_TAB_URI);
+        Preconditions.checkArgument(
+            StringUtils.isNotBlank(keyTabUri), "If you use Kerberos, key tab uri can't be blank");
+
+        try {
+          URI uri = new URI(keyTabUri);
+          String scheme = Optional.ofNullable(uri.getScheme()).orElse("file");
+          switch (scheme) {
+            case "http":
+            case "https":
+            case "ftp":
+              int timeout =
+                  (int)
+                      catalogPropertiesMetadata.getOrDefault(
+                          conf, HiveCatalogPropertiesMeta.FETCH_TIMEOUT_SEC);
+              URLConnection uc = uri.toURL().openConnection();
+              uc.setConnectTimeout(timeout * 1000);
+              uc.setReadTimeout(timeout * 1000);
+              uc.connect();
+
+              Files.copy(uc.getInputStream(), keyTabFile.toPath());
+              break;
+
+            case "file":
+              Files.createSymbolicLink(keyTabFile.toPath(), new File(uri.getPath()).toPath());
+              break;
+
+              // TODO: Supports to download keytab from HCFS, it's unsafe to store
+              // keytab in an unsecured HDFS cluster. If we use a secured HDFS cluster,
+              // we should have a keytab first.
+            default:
+              throw new IllegalArgumentException(
+                  String.format("Don't support the scheme %s", scheme));
+          }
+        } catch (URISyntaxException ue) {
+          throw new IllegalArgumentException("The uri of keytab has the wrong format", ue);
+        }
```

Comment:
Can you please move this to a common module with a generic method?

Reply:
Fixed.